tests/v2/validation/charts/backup_restore/backup_restore_test.go

```diff
@@ -0,0 +1,204 @@
+	"github.com/stretchr/testify/suite"
+)
+
+type BackupTestSuite struct {
+	suite.Suite
+	client              *rancher.Client
+	session             *session.Session
+	project             *management.Project
+	chartInstallOptions *charts.InstallOptions
+	chartFeatureOptions *charts.RancherBackupOpts
+	S3Client            *s3.Client
+}
+
+func (b *BackupTestSuite) TearDownSuite() {
+	b.session.Cleanup()
+}
+
+func (b *BackupTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	b.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(b.T(), err)
+
+	b.client = client
+
+	broConfig := new(charts.Config)
+	config.LoadConfig(charts.ConfigurationFileKey, broConfig)
+
+	project, err := projects.GetProjectByName(client, cluster, clusterProject)
+	require.NoError(b.T(), err)
+
+	// Get clusterName from config yaml
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(b.T(), clusterName, "Cluster name to install is not set")
+
+	// Get cluster meta
+	cluster, err := clusters.NewClusterMeta(client, clusterName)
+	require.NoError(b.T(), err)
+
+	// Get latest version of Rancher Backup chart
+	latestBackupVersion, err := client.Catalog.GetLatestChartVersion(backupChartName, catalog.RancherChartRepo)
+	require.NoError(b.T(), err)
+
+	b.chartInstallOptions = &charts.InstallOptions{
+		Cluster:   cluster,
+		Version:   latestBackupVersion,
+		ProjectID: project.ID,
+	}
+	b.chartFeatureOptions = &charts.RancherBackupOpts{
+		VolumeName:                broConfig.VolumeName,
+		BucketName:                broConfig.S3BucketName,
+		CredentialSecretName:      secretName,
+		CredentialSecretNamespace: broConfig.CredentialSecretNamespace,
+		Enabled:                   backupEnabled,
+		Endpoint:                  broConfig.S3Endpoint,
+		Folder:                    broConfig.S3FolderName,
+		Region:                    broConfig.S3Region,
+	}
+
+	subSession := b.session.NewSession()
+	defer subSession.Cleanup()
+}
+
+func (b *BackupTestSuite) TestBackupChart() {
+	subSession := b.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := b.client.WithSession(subSession)
+	require.NoError(b.T(), err)
+
+	b.client = client
+
+	broConfig := new(charts.Config)
+	config.LoadConfig(charts.ConfigurationFileKey, broConfig)
+
+	project, err := projects.GetProjectByName(b.client, cluster, clusterProject)
+	require.NoError(b.T(), err)
+
+	b.project = project
+
+	logrus.Info("Checking if the backup chart is already installed...")
+	initialBackupChart, err := charts.GetChartStatus(b.client, b.project.ClusterID, backupChartNamespace, backupChartName)
+	require.NoError(b.T(), err)
+
+	if !initialBackupChart.IsAlreadyInstalled {
+		installBroChart(b.T(), b.client, b.chartInstallOptions, b.chartFeatureOptions, b.project)
+	}
+
+	client, err = b.client.ReLogin()
+	require.NoError(b.T(), err)
+
+	logrus.Info("Creating two users...")
+	u1, err := users.CreateUserWithRole(client, users.UserConfig(), member)
```

Comment:
Can we change this to the following so that we can avoid repetitions here?
	userList   = []*management.User{}
	projList   = []*management.Project{}
	roleList   = []*management.RoleTemplate{}

logrus.Info("Creating two users, projects, and role templates...")

			for i := 0; i < 2; i++ {
				u, err := users.CreateUserWithRole(client, users.UserConfig(), roles[i])
				require.NoError(b.T(), err)
				userList = append(userList, u) 

				p, err := createProject(client, projects[i])
				require.NoError(b.T(), err)
				projList = append(projList, p)

				rt, err := createRole(client, roleContext, roleName, rules)
				require.NoError(b.T(), err)
				roleList = append(roleList, rt)
			}
We can move this to a helper as we have to make use of the same function later in the tests and then in that helper we can return the lists `userList, projList , roleList `  to be made available post backup is restored.